aggregations.ml

```diff
@@ -84,22 +85,26 @@ let analyze_single name agg_type json =
     | _ ->
     json,
     match agg_type with
-    | "max" | "min" -> Simple_metric (`MinMax, value ())
-    | "sum" -> Simple_metric (`Sum, value ())
+    | "max" | "min" -> Simple_metric (`MinMax, value json)
+    | "sum" -> Simple_metric (`Sum, value json)
     | "cumulative_sum" -> Cumulative_sum U.(get "buckets_path" to_string json)
-    | "avg" -> Simple_metric (`Avg, value ())
-    | "value_count" -> Value_count (value ())
-    | "cardinality" -> Cardinality (value ())
-    | "terms" -> Terms { term = value (); size = U.member "size" json }
-    | "significant_terms" -> Significant_terms { term = value (); size = U.member "size" json }
-    | "significant_text" -> Significant_text { term = value (); size = U.member "size" json }
-    | "histogram" -> Histogram (value ())
-    | "date_histogram" -> Date_histogram { on = value (); format = `Null <> U.member "format" json }
+    | "avg" -> Simple_metric (`Avg, value json)
+    | "value_count" -> Value_count (value json)
+    | "cardinality" -> Cardinality (value json)
+    | "weighted_avg" ->
+      let value_field = value (U.member "value" json) in
+      let weight_field = value (U.member "weight" json) in
```

Comment:
`weighted_avg` configuration holds field names in additional wrappers, not on the top level like others. To support that I changed `value` function to require `json` explicitly